src/components/RightSidebar/SearchMessages/SearchMessagesTab.vue

```diff
@@ -0,0 +1,442 @@
+		if (term.length === 0 && !fromUser.value && !sinceDate.value && !untilDate.value) {
+			return
+		}
+		const response = await request({
+			term: term.length !== 0 ? term : null,
+			person: fromUser.value?.id,
+			since: sinceDate.value?.toISOString(),
+			until: untilDate.value?.toISOString(),
+			limit: searchLimit.value,
+			cursor: searchCursor.value || null,
+			from: `/call/${token.value}`,
+		})
+
+		const data = response?.data?.ocs?.data
+		if (data?.entries.length > 0) {
+			let entries = data?.entries
+
+			isSearchExhausted.value = entries.length < searchLimit.value
+			searchCursor.value = data.cursor
+
+			// FIXME: remove the filter after the person filter is fixed on the server
+			if (fromUser.value) {
+				entries = entries.filter((entry) => entry.attributes.actorId === fromUser.value?.id)
+				if (entries.length === 0 && !isSearchExhausted.value) {
+					return await loadMore()
+				}
+			}
+
+			searchResults.value = searchResults.value.concat(entries.map((entry : Omit<MessageSearchResultEntry, 'to'>) => {
+				return {
+					...entry,
+					to: {
+						name: 'conversation',
+						hash: `#message_${entry.attributes.messageId}`,
+						params: {
+							token: entry.attributes.conversation,
+							skipLeaveWarning: true
+						}
+					}
+				}
+			})
+			)
+		}
+	} catch (exception) {
+		if (CancelableRequest.isCancel(exception)) {
+			return
+		}
+		console.error('Error searching for messages', exception)
+		showError(t('spreed', 'An error occurred while performing the search'))
+	} finally {
+		isFetchingResults.value = false
+	}
+}
+
+const debounceFetchSearchResults = debounce(fetchNewSearchResult, 250)
+</script>
+
+<template>
+	<div class="search-messages-tab">
+		<div class="search-form">
+			<div class="search-form__main">
+				<div class="search-form__search-box-wrapper">
+					<SearchBox ref="searchBox"
+						:value.sync="searchText"
+						:placeholder-text="t('spreed', 'Search messages …')"
+						:is-focused.sync="isFocused"
+						@input="debounceFetchSearchResults" />
+					<NcButton :pressed.sync="searchDetailsOpened"
+						:aria-label="t('spreed', 'Search options')"
+						:title="t('spreed', 'Search options')"
+						type="tertiary-no-background">
+						<template #icon>
+							<IconFilter :size="15" />
+						</template>
+					</NcButton>
+				</div>
+				<TransitionWrapper name="radial-reveal">
+					<div v-show="searchDetailsOpened" class="search-form__search-detail">
+						<NcSelect v-model="fromUser"
+							class="search-form__search-detail__from-user"
+							:aria-label-combobox="t('spreed', 'From User')"
+							:placeholder="t('spreed', 'From User')"
+							user-select
+							:loading="!participantsInitialised"
+							:options="participants"
+							@update:modelValue="debounceFetchSearchResults" />
+						<div class="search-form__search-detail__date-picker-wrapper">
+							<NcDateTimePickerNative id="search-form__search-detail__date-picker--since"
+								v-model="sinceDate"
```

Comment:
vue-library issue:
<img width="284" alt="image" src="https://github.com/user-attachments/assets/da741a86-3ff4-4423-ae39-48bc132e10f9" />

for now you can use `isNaN(sinceDate.value)` in code to catch it and replace with null